app/models/pipeline_run.rb

```diff
@@ -301,4 +302,25 @@ def count_unmapped_reads
     unidentified_fasta = get_s3_file(sample.unidentified_fasta_s3_path)
     unidentified_fasta.lines.select { |line| line.start_with? '>' }.count if unidentified_fasta
   end
+
+  delegate :project_id, to: :sample
+
+  def notify?
+    incomplete_runs = PipelineRun.where("id in (select max(id) from pipeline_runs group by sample_id) and sample_id in (select id from samples where project_id = #{project_id.to_i})").where("job_status != ?", PipelineRun::STATUS_CHECKED)
```

Comment:
I am not sure what this query does ;)

Reply:
For all samples in a project, check whether its most recent pipeline_run is still in progress. (If so, we won't want to notify the user that the project is complete yet.)

Reply:
I think incomplete_runs.count == 0 will probably make this query faster with less data transfered.